doc/design/doi-generation-2.md

```diff
@@ -33,10 +33,10 @@ DataCite allows for three types of DOIs ([DataCite](https://support.datacite.org
 We propose to:
 
 - Instead of a fake DOI, upon creation of a **public** dandiset, mint and use a legit `Draft DOI` `10.48324/dandi.{dandiset.id}` with
- - *minimal metadata* entered during creation request (title, description, license)
- - DLP URL `https://dandiarchive.org/dandiset/{dandiset.id}`
- - For embargoed dandiset, **do not** specify any metadata besides the DLP URL.
- - If minting a DOI fails, we need to raise exception to inform developers about the issue but proceed with the creation of the dandiset.
+  - *minimal metadata* entered during creation request (title, description, license)
+  - DLP URL `https://dandiarchive.org/dandiset/{dandiset.id}`
+  - For embargoed dandiset, **do not** specify any metadata besides the DLP URL.
```

Comment:
a thought: perhaps the doi generation process can be implemented only when dataset is unembargoed. i.e. embargoed datasets cannot be pointed to by doi (even if we get reviewer view only access). an owner would have to unembargo it. 

for implementation, doi generation happens at creation for public and unembargoing for embargoed dandisets.

Reply:
We could. But one of the goals here is to get away from using "fake DOIs" (#1709) which this would "prevent".

In current design (might change) we would make overall dandiset DOI Findable only upon initial publication. As embargoed dataset would never be published, its DOI would remain `Draft` thus not available to users, and thus IMHO there is no harm.  I think we could even keep updating it with metadata etc. That IMHO would simplify the logic and make "embargoed" less special (thus easier to code/troubleshoot etc).

Reply:
> In current design (might change) we would make overall dandiset DOI Findable only upon initial publication.

Given demand from zarr users to get DOIs for dandisets with zarrs, we better make them Findable as soon as possible in the life cycle of those dandisets. ref: https://github.com/dandi/helpdesk/discussions/165#discussioncomment-11824068 .  So I think we should proceed that way -- make them Findable as soon as datacite validation passes. Also inform user about datacite model issues as part of the validation.